plugins/microservices/src/get_server_property.cpp

```diff
@@ -0,0 +1,109 @@
+#include "irods/irods_server_properties.hpp"
+#include "irods/irods_ms_plugin.hpp"
+#include "irods/irods_re_structs.hpp"
+#include "irods/msParam.h"
+#include "irods/rodsErrorTable.h"
+#include "irods/irods_error.hpp"
+#include "irods/irods_logger.hpp"
+
+#include <exception>
+#include <functional>
+#include <string>
+
+namespace
+{
+    using log = irods::experimental::log;
+
+    auto msi_impl(msParam_t* _in_name, msParam_t* _out_value, ruleExecInfo_t* rei) -> int
+    {
+        RE_TEST_MACRO("    Calling msi_get_server_property");
+
+        if (!rei || !rei->rsComm) {
+            log::microservice::error("msi_get_server_property: input rei or rsComm is NULL.");
+            return SYS_INTERNAL_NULL_INPUT_ERR;
+        }
+
+        // ensure that this user is an admin
+        if (rei->uoic->authInfo.authFlag < LOCAL_PRIV_USER_AUTH) {
+            log::microservice::error("msi_get_server_property: User {} is not local admin. status = {}",
+                                     rei->uoic->userName,
+                                     CAT_INSUFFICIENT_PRIVILEGE_LEVEL);
+            return CAT_INSUFFICIENT_PRIVILEGE_LEVEL;
+        }
+
+        try {
+            std::string name = parseMspForStr(_in_name);
+            nlohmann::json json_value;
```

Comment:
Currently, we're copying the JSON object stored in the config object.

We can avoid that if this were a pointer.